src/scripts/cfo_supervisor.sh

```diff
@@ -10,10 +10,19 @@ git --version
 
 while true
 do
-    rm -rf ./tigerbeetle
+    # Drop the cache every ~24 hours.
+    if [ $((RANDOM % 24 )) -eq 0 ]
+    then rm -rf ./tigerbeetle
+    fi
+
     (
-        git clone https://github.com/tigerbeetle/tigerbeetle tigerbeetle
+        if ! [ -d ./tigerbeetle ]
+        then git clone https://github.com/tigerbeetle/tigerbeetle tigerbeetle
+        fi
+
         cd tigerbeetle
+        git fetch
+        git checkout -f origin/main
```

Comment:
GitHub ate my comment :(

Let's use `git switch --discard-changes`:

- as a rule of thumb, scripts should use full flags, like `--force` instead of `-f`, to make sure the reader doesn't have second guess whether it is force, file, follow, or something else
- `git switch` is a more safe replacement for `git checkout`. In checkout, the argument can be _either_ a file path or a comittish, and git decides heuristically which is which. `switch` only operates on comitishes.